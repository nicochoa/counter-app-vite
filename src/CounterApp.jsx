import { useState } from "react";
import {PropTypes} from "prop-types"

export const CounterApp = ( { value } ) => {
    // console.log(props);
    const[ counter, setCounter ] = useState(value)

    const handleAdd = ()=>{
        // console.log('+1')
        setCounter( counter + 1 );
        // setCounter( (c)=> c + 1 );
    }

    const handleSubstract = ()=>{
        setCounter( (c)=> c - 1 );
    }

    const handleReset = ()=>{
        setCounter( value );
    }

    return (
        <>
            <h1>Counter</h1>
            <h2> { counter } </h2>
            <button onClick={ handleAdd }> +1 </button>
            <button onClick={ handleSubstract }> -1 </button>
            <button onClick={ handleReset }> Reset </button>
        </>
    )
};

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}