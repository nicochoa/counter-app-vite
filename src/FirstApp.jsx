import { PropTypes } from "prop-types";


// usando funcion para pintar en el h1 
// const user = () => {
//     return 'Nicolas'
// }

// prueba para retornar objetos y se muestra como solo se puede especificando el key
// const newMessage = {
//     message: 'Hola mundo',
//     title: 'Fernando',
// };

export const FirstApp = ( { title, subTitle, name } ) => {
    // console.log(props);
    return (
        <>
            <h1 data-testid="test-title">{ title }</h1>
            {/* <code>{ JSON.stringify(newMessage) }</code> */}
            <p data-testid="test-subtitle"> { subTitle } </p>
            <p data-testid="test-subtitle"> { subTitle } </p>
            <p> { name } </p>
        </>
    )
};

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
}
FirstApp.defaultProps = {
    name: 'Nicolas Ochoa',
    subTitle: 'No hay subtitulo',
    // title: 'No hay titulo',
}
