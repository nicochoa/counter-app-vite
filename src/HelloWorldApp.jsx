// export function App() {
//     return (<h1>Hola mundo!!!</h1>);
// }

// export const hola = 123; 


export const HelloWorldApp = () => {
  return (
    <h1>Hello World App!!!</h1>
  );
}

// export const HelloWorldApp = () => <h1>Arrow Function!!!</h1>