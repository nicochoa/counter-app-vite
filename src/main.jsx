import React, { StrictMode } from 'react';
import ReactDOM from 'react-dom/client';

// import { HelloWorldApp } from './HelloWorldApp';
import { FirstApp } from './FirstApp';

import './style.css';
import { CounterApp } from './CounterApp';


ReactDOM.createRoot( document.getElementById('root')).render(
    <StrictMode>
        <CounterApp value={ 4 } />
    </StrictMode>
);