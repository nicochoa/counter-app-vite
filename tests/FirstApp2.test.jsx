import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en  <FirstApp/>', () => {
  const title = 'Hola soy Vegeta';
  const subTitle = 'Soy un subtitulo'


  test('debe hacer match con el snapshot', () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });
  test('debe mostrar el mensaje "Hola soy Goku"', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
    // screen.debug();
  });
  test('debe mostrar el titulo en un H1', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole('heading', { level: 1, }).innerHTML).toContain(title);
  });
  test('debe mostrar el subtitulo enviado por props', () => {
    render(<FirstApp title={title} subTitle={subTitle} />);
    expect(screen.getAllByText(subTitle).length).toBe(2);
  });
});