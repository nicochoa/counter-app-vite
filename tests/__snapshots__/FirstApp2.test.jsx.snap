// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en  <FirstApp/> debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy Vegeta
  </h1>
  <p
    data-testid="test-subtitle"
  >
     
    No hay subtitulo
     
  </p>
  <p
    data-testid="test-subtitle"
  >
     
    No hay subtitulo
     
  </p>
  <p>
     
    Nicolas Ochoa
     
  </p>
</div>
`;
